### ISSUE TITLE
Incorrect height of CollectionView when ItemsSource is empty

### ISSUE THREAD

**Issue by domitorosan** [dotnet/maui#23148]
### Description

If CollectionView has 'Auto' height (either is nested in VerticalStackLayout, has VerticalOptions="Start", or is nested in a Grid with Auto row height), has specified ItemsLayout and empty ItemsSource, it expands to giant sizes.

The following example displays a rendered height of 6,100,797.82 pixels:

![image](https://github.com/dotnet/maui/assets/152199498/d6e0f929-dc25-4181-b502-bc5e8c9989ba)

If you add at least one item to Collection in ItemsSource, it starts behaving as expected, adjusting its height to its content.

![image](https://github.com/dotnet/maui/assets/152199498/0dbee5fb-8614-4630-a54e-6b3063e4a0df)

### Steps to Reproduce

1. Create a new MAUI project
2. Put CollectionView into any container with Auto height (like VerticalStackLayout)
3. Specify Vertical CollectionView.ItemsLayout
4. Enjoy the show.

Here is my code of View:
```
<ContentPage xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
			 xmlns:main="clr-namespace:CollectionViewBug"
             x:Class="CollectionViewBug.MainPage"
			 x:DataType="main:MainPage">

	<VerticalStackLayout Padding="30,0" Spacing="25">
		
		<Button
			x:Name="CounterBtn"
			Text="Click me" 
			SemanticProperties.Hint="Counts the number of times you click"
			Command="{Binding AddCommand}"
			HorizontalOptions="Fill" />
		
		<CollectionView ItemsSource="{Binding Items}" Background="Green">
			<CollectionView.ItemsLayout>
				<LinearItemsLayout Orientation="Vertical" ItemSpacing="8"/>
			</CollectionView.ItemsLayout>
			<CollectionView.ItemTemplate>
				<DataTemplate>
					<Label Text="{Binding}" />
				</DataTemplate>
			</CollectionView.ItemTemplate>
		</CollectionView>
		
		<Label
			Text="Bottom text"
			Style="{StaticResource Headline}"
			SemanticProperties.HeadingLevel="Level1" />
		
	</VerticalStackLayout>
</ContentPage>
```

And the code-behind:

```
using System.Collections.ObjectModel;
using System.Windows.Input;

namespace CollectionViewBug
{
	public partial class MainPage : ContentPage
	{
		public ObservableCollection<string> Items { get; set; } = [];

		public ICommand AddCommand => new Command(() => Items.Add("Item"));

		public MainPage()
		{
			InitializeComponent();
			BindingContext = this;
		}
	}
}
```

The code was made as a quick example, so please excuse me if its architecture is imperfect.

### Version with bug

8.0.7 SR2

### Is this a regression from previous behavior?

Not sure, did not test other versions

### Last version that worked well

Unknown/Other

### Affected platforms

Android, I was *not* able test on other platforms

### Affected platform versions

Android 34.0.52

**Comment by RoiChen001** [comment]
I can repro this at Android platform on the latest 17.11.0 Preview 2.0(8.0.7&8.0.60).

**Comment by drasticactions** [comment]
I don't believe this is a bug. I think this was a breaking change with [StackLayout](https://learn.microsoft.com/en-us/dotnet/maui/migration/layouts?view=net-maui-8.0#stacklayout) introduced with MAUIs introduction with net6.0.

StackLayout does not automatically expand elements inside it. That includes CollectionView. If you want this to work, you would change your outer StackLayout to a Grid (With something like `Auto, *, Auto` for the Row Layout).

@PureWeen What do you think?

**Comment by domitorosan** [comment]
> I don't believe this is a bug. I think this was a breaking change with [StackLayout](https://learn.microsoft.com/en-us/dotnet/maui/migration/layouts?view=net-maui-8.0#stacklayout) introduced with MAUIs introduction with net6.0.
> 
> StackLayout does not automatically expand elements inside it. That includes CollectionView. If you want this to work, you would change your outer StackLayout to a Grid (With something like `Auto, *, Auto` for the Row Layout).
> 
> @PureWeen What do you think?

I didn't expect StackLayout to expand elements inside it. The intention is the opposite—to shrink CollectionView to its content size. As I mentioned, if you add at least one item it starts acting as expected adjusting the height to its content. 
When a CollectionView in a StackLayout has no items, I expect it to have 0 height. And I don't think that under any circumstances the height of anything has to be 6+ million.

**Comment by PureWeen** [comment]
yea, ultimately I'd say @drasticactions is right, but, it seems like in this scenario the dimensions of the CV should be zero since there's not an emptyitemtemplate

**Comment by domitorosan** [comment]
Also, if you add EmptyView it doesn't help either. You see EmptyView in that case, yet the height remains over millions. That was how I discovered the issue in the first place, I expected CollectionView to display EmptyView fitting to its size, not rocketing the height into the sky.

**Comment by Felicity-R** [comment]
I'm working around this with 
```
<CollectionView.Triggers>
  <DataTrigger TargetType="CollectionView" Binding="{Binding Items, Converter={StaticResource EmptyListConverter}}" Value="True"> 
    <Setter Property="HeightRequest" Value="20"/>
  </DataTrigger>
</CollectionView.Triggers>
```
Where EmptyListConverter is the IsListNullOrEmptyConverter from the Maui Community Toolkit, and 20 is the desired height of my empty view.

**Comment by Felicity-R** [comment]
Unfortunately my workaround isn't actually working so well, because sometimes the CollectionView stays stuck at 20 height instead of growing back to the full height when the list is populated :/
Maybe someone else has a better suggestion.

**Comment by domitorosan** [comment]
I have tested it for `.NET 9 MAUI`, `Microsoft.Maui.Controls` version `9.0.90` (the latest available at the moment), and can confirm that the issue **is still present and relevant** for both `Debug` and `Release` modes.

To reproduce the issue, I followed the instructions provided in the description. Please note that I used an Android emulator (Android 14) for testing, and the results may differ from those on a real device or newer versions of Android.

Here is the screenshot of what it looks like:

<img width="360" height="780" alt="Image" src="https://github.com/user-attachments/assets/9d5fcdf8-57b6-40ac-8ada-8b529a0fcb48" />

**Comment by corne-ac** [comment]
Still getting this exact issue, on 10.0.70. Quite an old bug to still be active. Cant find a clean workaround as of yet.
